operator/controllers/kyma_controller.go

```diff
@@ -348,18 +372,63 @@ func (r *KymaReconciler) GetUnstructuredResource(ctx context.Context, gvr schema
 // SetupWithManager sets up the controller with the Manager.
 func (r *KymaReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.Builder = ctrl.NewControllerManagedBy(mgr)
-	return r.Builder.For(&operatorv1alpha1.Kyma{}).
-		//Owns(&corev1.ConfigMap{}).
-		Watches(
-			&source.Kind{Type: &corev1.ConfigMap{}},
-			handler.EnqueueRequestsFromMapFunc(r.findObjectsForConfigMap),
-			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
-		).
-		Complete(r)
+	return r.Builder.For(&operatorv1alpha1.Kyma{}).Complete(r)
 }
 
-func (r *KymaReconciler) findObjectsForConfigMap(configMap client.Object) []reconcile.Request {
-	requests := make([]reconcile.Request, 0)
-	// add code here to watch for changes in component mapping ConfigMap
-	return requests
+func (r *KymaReconciler) ComponentChangeHandler(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	objectBytes, err := json.Marshal(e.ObjectNew)
+	if err != nil {
+		panic(err)
+	}
+	componentObj := unstructured.Unstructured{}
+	if err = json.Unmarshal(objectBytes, &componentObj); err != nil {
+		panic(err)
+	}
+	if componentObj.Object["status"] == nil {
+		return
+	}
+	for key, value := range componentObj.Object["status"].(map[string]interface{}) {
+		if key == "state" && value == operatorv1alpha1.KymaStateReady {
+			kymaObj := operatorv1alpha1.Kyma{}
+			ownerRefs := componentObj.GetOwnerReferences()
+			var ownerName string
+
+			for _, ownerRef := range ownerRefs {
+				if kymaObj.Kind == ownerRef.Kind {
+					ownerName = ownerRef.Name
+					break
+				}
+			}
+
+			if err := r.Get(context.TODO(), types.NamespacedName{Name: ownerName, Namespace: componentObj.GetNamespace()}, &kymaObj); err != nil {
+				return
+			}
+
+			kymaObj.Status.State = operatorv1alpha1.KymaStateReady
+			kymaObj.Status.ObservedGeneration = kymaObj.Generation
+			var condition *operatorv1alpha1.KymaCondition
+
+			for _, existingCondition := range kymaObj.Status.Conditions {
+				if existingCondition.Type == operatorv1alpha1.ConditionTypeReady {
+					condition = &existingCondition
```

Comment:
this loop is always taking the last condition right now, I think we could make it break here. The way I see it every state will have one condition. so checking on kymaObj.Status.Conditions could mean 3-4 conditions. That means we will have to find a way to get the right condition. I think we need the technical reason message to be mapped to the reconciler in this case

Reply:
Every state cannot have one condition each IMO. 
I thought we discussed on having one Ready condition / component

Reply:
exactly but if we have one ready condition / component then we will find here 5 conditions for 5 components in one kyma status. Right now the logic would exit with the last ready condition. If we have the setup of 3 conditions with the conditions set to "ready, not ready, ready", and you wanted to change the second condition, then you would only change the last one to ready, not causing any change, no?